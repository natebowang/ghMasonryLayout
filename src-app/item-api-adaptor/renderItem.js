import React from 'react';

export const titlePaddingRem = 0.4;
export const titleFontSizeRem = 1.2;
export const titleLineHeight = 1.15;
const titleStyle = {
    padding: titlePaddingRem + 'rem',
    fontSize: titleFontSizeRem + 'rem',
    lineHeight: titleLineHeight,
    fontWeight: 'bold',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    wordBreak: 'break-word',
    hyphens: 'auto',
    overflow: 'hidden',
};
export const contentPaddingRem = 0.4;
export const contentFontSizeRem = 1;
export const contentLineHeight = 1.15;
const contentStyle = {
    padding: contentPaddingRem + 'rem',
    fontSize: contentFontSizeRem + 'rem',
    lineHeight: contentLineHeight,
    borderRadius: '0.2rem',
    background: '#e0f0f0',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    wordBreak: 'break-word',
    overflow: 'hidden',
};

const renderItem = (item) => {
    return (
        <>
            <div style={titleStyle}>
                {item.id} {item.title}
            </div>
            <div style={contentStyle}>
                {item.content}
            </div>
        </>
    )
};
export default renderItem;