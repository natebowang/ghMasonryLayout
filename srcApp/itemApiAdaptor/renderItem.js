import React from 'react';

export const titlePadding = 0.4;
export const titleFontSize = 1.2;
export const titleLineHeight = 1.15;
const titleStyle = {
    padding: titlePadding + 'rem',
    fontSize: titleFontSize + 'rem',
    lineHeight: titleLineHeight,
    fontWeight: 'bold',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    wordBreak: 'break-word',
    hyphens: 'auto',
    overflow: 'hidden',
};
export const contentPadding = 0.4;
export const contentFontSize = 1;
export const contentLineHeight = 1.15;
const contentStyle = {
    padding: contentPadding + 'rem',
    fontSize: contentFontSize + 'rem',
    lineHeight: contentLineHeight,
    borderRadius: '0.2rem',
    background: '#e0f0f0',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    wordBreak: 'break-word',
    overflow: 'hidden',
};

const renderItem = (item) => {
    return (
        <>
            <div style={titleStyle}>
                {item.id} {item.title}
            </div>
            <div style={contentStyle}>
                {item.content}
            </div>
        </>
    )
};
export default renderItem;