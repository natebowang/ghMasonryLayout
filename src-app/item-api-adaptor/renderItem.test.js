import React from 'react';
import renderer from 'react-test-renderer';
import renderItem from './renderItem';

const item = {
    "id": 1,
    "title": "HTTP headers for the responsible developer",
    "content": "A hidden Cupid in Vermeer’s Girl Reading a Letter at an Open Window, one of the world’s most famous paintings, is set to resurface on the canvas after two and a half centuries behind a layer of paint. During restoration work, conservators discovered, to their surprise, that the naked figure—which dominates the upper right section of the picture—was overpainted long after the artist’s death."
};

const MockContainer = ({renderItem, item}) => {
    return (
        <>
            {renderItem(item)}
        </>
    )
};

// snapshot test with react-test-renderer
test("App changes the class when hovered", () => {
    const component = renderer.create(
        <MockContainer renderItem={renderItem}
                       item={item}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
