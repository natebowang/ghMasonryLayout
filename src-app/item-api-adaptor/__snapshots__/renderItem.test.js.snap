// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App changes the class when hovered 1`] = `
Array [
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontSize": "1.2rem",
        "fontWeight": "bold",
        "hyphens": "auto",
        "justifyContent": "center",
        "lineHeight": 1.15,
        "overflow": "hidden",
        "padding": "0.4rem",
        "wordBreak": "break-word",
      }
    }
  >
    1
     
    HTTP headers for the responsible developer
  </div>,
  <div
    style={
      Object {
        "alignItems": "center",
        "background": "#e0f0f0",
        "borderRadius": "0.2rem",
        "display": "flex",
        "fontSize": "1rem",
        "justifyContent": "center",
        "lineHeight": 1.15,
        "overflow": "hidden",
        "padding": "0.4rem",
        "wordBreak": "break-word",
      }
    }
  >
    A hidden Cupid in Vermeer’s Girl Reading a Letter at an Open Window, one of the world’s most famous paintings, is set to resurface on the canvas after two and a half centuries behind a layer of paint. During restoration work, conservators discovered, to their surprise, that the naked figure—which dominates the upper right section of the picture—was overpainted long after the artist’s death.
  </div>,
]
`;
